chalk-solve/src/solve/slg/aggregate.rs

```diff
@@ -425,7 +434,53 @@ impl<I: Interner> AntiUnifier<'_, '_, I> {
         }
     }
 
-    fn new_variable(&mut self) -> Ty<I> {
+    fn aggregate_consts(&mut self, c1: &Const<I>, c2: &Const<I>) -> Const<I> {
+        let interner = self.interner;
+
+        // FIXME actually check that c1 and c2 have the same type
```

Comment:
Is there a reason we can't invoke `let c_ty = self.aggregate_types(..)` or something like that? 

Well, I guess that's not quite what we want probably. We don't want to replace the types with inference variables. I guess that Rust's type system generally guarantees that the types here will be equivalent, so this is more of a "validation" check, right? (In which case, this isn't really a *FIXME* so much as "hmm this would be nice"?)

Reply:
Hmm, you are right, I'll change the wording then